src/Factory/BaseFactory.php

```diff
@@ -118,25 +120,23 @@ abstract protected function getRootTableRegistryName(): string;
     abstract protected function setDefaultTemplate(): void;
 
     /**
-     * @param array|callable|null|int|\Cake\Datasource\EntityInterface|string $makeParameter Injected data
+     * @param \Cake\Datasource\EntityInterface|callable|array|string|int|null $makeParameter Injected data
      * @param int                     $times Number of entities created
      * @return static
      */
-    public static function make($makeParameter = [], int $times = 1): BaseFactory
-    {
+    public static function make(
+        array|callable|int|EntityInterface|string|null $makeParameter = [],
+        int $times = 1
+    ): BaseFactory {
         if (is_numeric($makeParameter)) {
             $factory = self::makeFromNonCallable();
             $times = $makeParameter;
         } elseif (is_null($makeParameter)) {
             $factory = self::makeFromNonCallable();
         } elseif (is_array($makeParameter) || $makeParameter instanceof EntityInterface || is_string($makeParameter)) {
             $factory = self::makeFromNonCallable($makeParameter);
-        } elseif (is_callable($makeParameter)) {
-            $factory = self::makeFromCallable($makeParameter);
         } else {
-            throw new InvalidArgumentException('
-                ::make only accepts an array, an integer, an EntityInterface, a string or a callable as first parameter.
-            ');
+            $factory = self::makeFromCallable($makeParameter);
```

Comment:
Don't know if you want to keep it this to require a specific param type or prefer the manually thrown exception

Reply:
I like throwing a specific exception with a verbose message so the user learns how to use the factories.

Reply:
Then I will change the param type to mixed and re-add the exception